### PR TITLE
fix missing dependency determination

### PR DESCRIPTION
**Pull request by depombo**

### CODE REVIEW

File: src/services/iasql.ts [hunk 1]
```diff
@@ -713,11 +713,12 @@ export async function install(moduleList: string[], dbAlias: string, user: any,
     }`);
   }
   // Check to make sure that all dependent modules are in the list
-  const missingDeps = mods.map((m: Modules.ModuleInterface) => m.dependencies.find(d => !moduleList.includes(d)));
```
Comment (depombo) [inline src/services/iasql.ts:716]: `find` only returns the first value that matches the predicate so this wasn't returning a full list

Comment (aguillenv) [reply src/services/iasql.ts:716]: hmm curious how it was working or why it was done with the `find`?

**Review by aguillenv — APPROVED** [review]

File: src/services/iasql.ts (hunk 1 above)
Comment (depombo) [reply src/services/iasql.ts:716]: it was always returning `aws_account` which was the first match and that was then filtered so it was a always a noop

File: src/services/iasql.ts [hunk 2]
```diff
@@ -731,6 +723,16 @@ export async function install(moduleList: string[], dbAlias: string, user: any,
       i--;
     }
   }
+  // TODO rm special casing for aws_account
+  // Check to make sure that all dependent modules are in the list
+  const missingDeps = mods
+    .flatMap((m: Modules.ModuleInterface) => m.dependencies.filter(d => !moduleList.includes(d) && !existingModules.includes(d)))
```
Comment (depombo) [inline src/services/iasql.ts:729]: added the check for `existingModules`

Comment (aguillenv) [reply src/services/iasql.ts:729]: meaning the ones are already installed, right?

Comment (depombo) [reply src/services/iasql.ts:729]: yes

**Review by aguillenv — APPROVED** [review]

**Pull request merged**